cellprofiler/modules/measureimageskeleton.py

```diff
@@ -0,0 +1,156 @@
+import cellprofiler.module
+import cellprofiler.setting
+import numpy
+import scipy.ndimage
+import skimage.measure
+import skimage.segmentation
+import skimage.util
+
+
+def _neighbors(image):
+    padding = skimage.util.pad(image, 1, "constant")
+
+    mask = padding > 0
+
+    padding = padding.astype(numpy.float)
+
+    if image.ndim == 2:
+        response = 3 ** 2 * scipy.ndimage.uniform_filter(padding) - 1
+
+        labels = (response * mask)[1:-1, 1:-1]
+
+        return labels.astype(numpy.uint16)
+    elif image.ndim == 3:
+        response = 3 ** 3 * scipy.ndimage.uniform_filter(padding) - 1
+
+        labels = (response * mask)[1:-1, 1:-1, 1:-1]
+
+        return labels.astype(numpy.uint16)
+
+
+def branches(image):
+    return _neighbors(image) > 2
+
+
+def endpoints(image):
+    return _neighbors(image) == 1
+
+
+class MeasureImageSkeleton(cellprofiler.module.Module):
```

Comment:
Additionally, you're going to need to implement a few more methods. According to the [module documentation](https://github.com/CellProfiler/CellProfiler/blob/8472a4b0b04879169b0a90b67a9e216658d53d5d/cellprofiler/module.py#L35-L40), you need to implement:

    get_categories - The category of measurement produced, for instance AreaShape
    get_measurements - The measurements produced by a category
    get_measurement_images - The images measured for a particular measurement
    get_measurement_scales - the scale at which a measurement was taken
    get_measurement_columns - the measurements stored in the database